c_src/fast_tls.c

```diff
@@ -1216,6 +1216,45 @@ static ERL_NIF_TERM get_negotiated_cipher_nif(ErlNifEnv *env, int argc,
     return enif_make_binary(env, &bin);
 }
 
+static ERL_NIF_TERM tls_get_peer_finished_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    state_t *state = NULL;
+    if (!enif_get_resource(env, argv[0], tls_state_t, (void *) &state))
+        return enif_make_badarg(env);
+
+    ErlNifBinary    bin;
+	/*
+	 * OpenSSL does not offer an API to directly get the length of the
+	 * expected TLS Finished message, so just do a dummy call to grab this
+	 * information to allow caller to do an allocation with a correct size.
+	 */
+    size_t len = SSL_get_peer_finished(state->ssl, NULL, 0);
+    if(!enif_alloc_binary(len, &bin))
+        return ERR_T(enif_make_atom(env, "enomem"));
+
+	(void) SSL_get_peer_finished(state->ssl, bin.data, len);
+    if (!bin.data)
```

Comment:
I don't think you need this? This call can't change value of this pointer, did you wanted to check what in data pointed by that pointer? Anyway if you wanted to return early you would also need to call enif_release_binary.

In general maybe you should use enif_make_new_binary, i think this is better match there.

Reply:
Actually, that's a very good point, such a classic error, forgetting a release after an allocation! `enif_make_new_binary` seems fine, I'll change it to that.